test/saithrift/src/switch_sai_rpc_server.cpp

```diff
@@ -1034,6 +1040,84 @@ class switch_sai_rpcHandler : virtual public switch_sai_rpcIf {
+bool sai_thrift_set_if_state(const std::string& if_name, bool if_state) {
+    std::string cmd;
+    cmd = std::string("ifconfig ") + if_name + ((if_state) ? std::string(" up"):std::string(" down"));
+    int ret = system(cmd.c_str());
+    return true;
+}
+
+sai_thrift_object_id_t sai_thrift_get_cpu_port_id() {
+      sai_status_t status;
+      sai_attribute_t attr;
+      sai_switch_api_t *switch_api;
+      sai_thrift_object_id_t cpu_port_id;
+      const char* f_name = __FUNCTION__;
+      printf("%s\n", f_name);
+      status = sai_api_query(SAI_API_SWITCH, (void **) &switch_api);
+      if (status != SAI_STATUS_SUCCESS) {
+          printf("%s failed to obtain switch_api, status:%d\n", f_name, status);
+          return SAI_NULL_OBJECT_ID;
+      }
+      attr.id = SAI_SWITCH_ATTR_CPU_PORT;
+      status = switch_api->get_switch_attribute(1, &attr);
+      if (status != SAI_STATUS_SUCCESS)
+      {
+          printf("%s failed, status:%d\n", f_name, status);
+          return SAI_NULL_OBJECT_ID;
+      }
+      cpu_port_id = (sai_thrift_object_id_t) attr.value.oid;
+      return cpu_port_id;
+  }
+
+  sai_thrift_object_id_t sai_thrift_get_default_router_id() {
+      sai_status_t status;
+      sai_attribute_t attr;
+      sai_switch_api_t *switch_api;
+      sai_thrift_object_id_t default_router_id;
+      const char* f_name = __FUNCTION__;
+      printf("%s\n", f_name);
+      status = sai_api_query(SAI_API_SWITCH, (void **) &switch_api);
+      if (status != SAI_STATUS_SUCCESS) {
+          printf("%s failed to obtain switch_api, status:%d\n", f_name, status);
+          return SAI_NULL_OBJECT_ID;
+      }
+      attr.id = SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID;
+      status = switch_api->get_switch_attribute(1, &attr);
+      if (status != SAI_STATUS_SUCCESS)
+      {
+          printf("%s. Failed to get switch virtual router ID, status %d", f_name, status);
+          return SAI_NULL_OBJECT_ID;
+      }
+      default_router_id = (sai_thrift_object_id_t)attr.value.oid;
+      return default_router_id;
+  }
+
+  sai_thrift_object_id_t sai_thrift_get_default_trap_group() {
+      sai_status_t status;
+      sai_attribute_t attr;
+      sai_switch_api_t *switch_api;
+      sai_thrift_object_id_t default_trap_group;
+      const char* f_name = __FUNCTION__;
+      printf("%s\n", f_name);
+      status = sai_api_query(SAI_API_SWITCH, (void **) &switch_api);
+      if (status != SAI_STATUS_SUCCESS) {
+          printf("%s failed to obtain switch_api, status:%d\n", f_name, status);
+          return SAI_NULL_OBJECT_ID;
+      }
+      attr.id = SAI_SWITCH_ATTR_DEFAULT_TRAP_GROUP;
+      status = switch_api->get_switch_attribute(1, &attr);
+      if (status != SAI_STATUS_SUCCESS)
+      {
+          printf("%s. Failed to get switch default trap group, status %d", f_name, status);
+          return SAI_NULL_OBJECT_ID;
+      }
+      default_trap_group = (sai_thrift_object_id_t)attr.value.oid;
+      return default_trap_group;
+  }
+
```

Comment:
just curious, why not to fill these attributes in sai_thrift_get_switch_attribute(x) instead of create new RPC API for 
each kind of attribute ?